### DISCUSSION TITLE
libluajit.dylib: No such file or directory

### DISCUSSION THREAD

**Discussion by carlosbaraza** [obsproject/obs-studio#10714] [Development Questions] [answered]
I am a complete beginner in the codebase. I am trying to build it locally in MacOS (arm64). However, I am getting the following error during the build:

```
obs-studio/build_macos/UI/Debug/OBS.app/Contents/Frameworks/libluajit.dylib: No such file or directory
Command CodeSign failed with a nonzero exit code
```

I am obviously missing a library. But I am not sure why or how should it be resolved.

**Comment by Fenrirthviti** [comment, marked as answer]
This sounds more likely to be a support request, and we are not currently accepting support requests on GitHub Discussions. Make sure you read any [pinned posts](https://github.com/obsproject/obs-studio/discussions/5206) on discussion boards before opening a new discussion. Please use our [forums](https://obsproject.com/forum/) or [Discord](https://obsproject.com/discord/) for further assistance.

Thank you!

**Reply by carlosbaraza** [reply]
If those are the rules, I am sorry. But future developers with the same problem will miss out on the discussion because it is buried in some discord chat.

**Reply by carlosbaraza** [reply]
For anyone that happened to land here with the same problem, this issue was discussed on discord: https://discord.com/channels/348973006581923840/374636084883095554/1177407747558473809

**Comment by RytoEX** [comment]
This is most likely caused by CMake finding LuaJIT via Homebrew, which can be confirmed by checking your CMakeCache.txt file. It probably said something like this:
```
//Luajit location
Luajit_LIBRARY:FILEPATH=/opt/homebrew/Cellar/luajit/2.1.1699180677/lib/libluajit.dylib
```

You have to remove or unlink that one so CMake can't find it and finds the one we provide.

**Comment by jcm93** [comment]
Is there something OBS can do to override/always prefer its own `libluajit.dylib` over a system/homebrew one? It is slightly annoying encountering this issue anytime one creates a new OBS Xcode project, as lots of software depends on luajit and entirely unlinking it is not ideal.

It also seems a bit odd that the .dylib that OBS ends up bundling with a homebrew `libluajit` is an alias pointing at itself. The alias`libluajit.dylib` inside the brew Cellar folder correctly points at `/opt/homebrew/Cellar/luajit/2.1.1713773202/lib/libluajit-5.1.2.1.1713773202.dylib`(version string current as of writing) but for some reason the one that ends up in OBS does not point there. Not sure why this behavior is unique to `libluajit` and not other system libraries that seem? to get used like mbedtls or x264.

**Reply by RytoEX** [reply]
As far as I know, we already try to. This was mentioned in the linked Discord conversation.
